watch_service.go

```diff
@@ -30,15 +30,10 @@ type watchService service
 
 // WatchResult represents a result from watch operation.
 type WatchResult struct {
-	Commit Commit
-	Entry  Entry
-	Res    *http.Response
-	Err    error
-}
-
-type commitWithEntry struct {
-	Commit
-	Entry Entry `json:"entry,omitempty"`
+	Revision int   `json:"revision"`
+	Entry    Entry `json:"entry,omitempty"`
+	Res      *http.Response
```

Comment:
I wonder if we could remove httpResponse from user view because:
1. User often not care about http status code
2. User won't read http response body because it was closed and was read
3. We already handled error
The lack from WatchResult I think is `StatusCode`. So how about add it and remove `Res`

Reply:
`HttpStatusCode` sounds good to me.

Reply:
That's a good suggestion. I just contained the `*http.Response` following https://github.com/google/go-github, but I've never seen it's really used.
Before removing it, I have a question. We do include the `*http.Response` in other APIs as well. Do you think we need to replace all of them with the HTTP status code?

Reply:
Yes, I think replace all as we already handle the error well.

Reply:
All right~! Let me do it in another PR. Thanks~!

Reply:
Yeah. So I think it's ok to merge this PR. I am looking forward to your next PR @minwoox 👍